### ISSUE TITLE
pwntools installation error

### ISSUE THREAD

**Issue by InfamousSYN** [InfamousSYN/CTF-Cookbook#8]
pwntools being stabled either by "node['pwntools']['package']" or by "node['pwntools']['latest']" does not install correctly due to missing dependencies.

**Comment by InfamousSYN** [comment]
[2015-06-01T10:35:32+00:00] ERROR: execute[Installing Pwntools] (CTF-Cookbook::ctf-tools line 74) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of pip install pwntools ----
STDOUT: Downloading/unpacking pwntools
  Running setup.py (path:/tmp/pip_build_root/pwntools/setup.py) egg_info for package pwntools
    
Downloading/unpacking paramiko (from pwntools)
Requirement already satisfied (use --upgrade to upgrade): argparse in /usr/lib/python2.7 (from pwntools)
Downloading/unpacking mako (from pwntools)
  Running setup.py (path:/tmp/pip_build_root/mako/setup.py) egg_info for package mako
    
    warning: no files found matching '*.xml' under directory 'examples'
    warning: no files found matching '*.mako' under directory 'examples'
    warning: no files found matching 'ez_setup.py'
    no previously-included directories found matching 'doc/build/output'
Downloading/unpacking pyelftools (from pwntools)
  Running setup.py (path:/tmp/pip_build_root/pyelftools/setup.py) egg_info for package pyelftools
    
Downloading/unpacking capstone==2.1 (from pwntools)
  Downloading capstone-2.1.tar.gz
  Running setup.py (path:/tmp/pip_build_root/capstone/setup.py) egg_info for package capstone
    
Downloading/unpacking ropgadget (from pwntools)
  Downloading ROPGadget-5.4.tar.gz
  Running setup.py (path:/tmp/pip_build_root/ropgadget/setup.py) egg_info for package ropgadget
    
Requirement already satisfied (use --upgrade to upgrade): pyserial in /usr/lib/python2.7/dist-packages (from pwntools)
Requirement already satisfied (use --upgrade to upgrade): requests in /usr/lib/python2.7/dist-packages (from pwntools)
Downloading/unpacking psutil (from pwntools)
  Running setup.py (path:/tmp/pip_build_root/psutil/setup.py) egg_info for package psutil
    
    warning: no previously-included files matching '*' found under directory 'docs/_build'
Requirement already satisfied (use --upgrade to upgrade): pycrypto>=2.1,!=2.4 in /usr/lib/python2.7/dist-packages (from paramiko->pwntools)
Downloading/unpacking ecdsa>=0.11 (from paramiko->pwntools)
Downloading/unpacking MarkupSafe>=0.9.2 (from mako->pwntools)
  Downloading MarkupSafe-0.23.tar.gz
  Running setup.py (path:/tmp/pip_build_root/MarkupSafe/setup.py) egg_info for package MarkupSafe
    
Installing collected packages: pwntools, paramiko, mako, pyelftools, capstone, ropgadget, psutil, ecdsa, MarkupSafe
  Running setup.py install for pwntools
    
    Installing unhex script to /usr/local/bin
    Installing elfpatch script to /usr/local/bin
    Installing constgrep script to /usr/local/bin
    Installing cyclic script to /usr/local/bin
    Installing hex script to /usr/local/bin
    Installing phd script to /usr/local/bin
    Installing elfdiff script to /usr/local/bin
    Installing shellcraft script to /usr/local/bin
    Installing disasm script to /usr/local/bin
    Installing asm script to /usr/local/bin
  Running setup.py install for mako
    
    warning: no files found matching '*.xml' under directory 'examples'
    warning: no files found matching '*.mako' under directory 'examples'
    warning: no files found matching 'ez_setup.py'
    no previously-included directories found matching 'doc/build/output'
    Installing mako-render script to /usr/local/bin
  Running setup.py install for pyelftools
    changing mode of build/scripts-2.7/readelf.py from 644 to 755
    
    changing mode of /usr/local/bin/readelf.py to 755
  Running setup.py install for capstone
    
  Running setup.py install for ropgadget
    changing mode of build/scripts-2.7/ROPgadget from 644 to 755
    
    changing mode of /usr/local/bin/ROPgadget to 755
  Running setup.py install for psutil
    building '_psutil_linux' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPSUTIL_VERSION=221 -I/usr/include/python2.7 -c psutil/_psutil_linux.c -o build/temp.linux-x86_64-2.7/psutil/_psutil_linux.o
    psutil/_psutil_linux.c:12:20: fatal error: Python.h: No such file or directory
     #include <Python.h>
                        ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/psutil/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-CbWtQT-record/install-record.txt --single-version-externally-managed --compile:
    running install

running build

running build_py

creating build

creating build/lib.linux-x86_64-2.7

creating build/lib.linux-x86_64-2.7/psutil

copying psutil/_pswindows.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_pssunos.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_pslinux.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_psbsd.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_psposix.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_common.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_psosx.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_compat.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/__init__.py -> build/lib.linux-x86_64-2.7/psutil

running build_ext

building '_psutil_linux' extension

creating build/temp.linux-x86_64-2.7

creating build/temp.linux-x86_64-2.7/psutil

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPSUTIL_VERSION=221 -I/usr/include/python2.7 -c psutil/_psutil_linux.c -o build/temp.linux-x86_64-2.7/psutil/_psutil_linux.o

psutil/_psutil_linux.c:12:20: fatal error: Python.h: No such file or directory

 #include <Python.h>

                    ^

compilation terminated.

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/psutil/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-CbWtQT-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip_build_root/psutil
Storing debug log for failure in /root/.pip/pip.log
STDERR: 
---- End output of pip install pwntools ----
Ran pip install pwntools returned 1

**Comment by InfamousSYN** [comment]
```
[2015-06-01T10:35:32+00:00] ERROR: execute[Installing Pwntools] (CTF-Cookbook::ctf-tools line 74) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of pip install pwntools ----
STDOUT: Downloading/unpacking pwntools
  Running setup.py (path:/tmp/pip_build_root/pwntools/setup.py) egg_info for package pwntools

Downloading/unpacking paramiko (from pwntools)
Requirement already satisfied (use --upgrade to upgrade): argparse in /usr/lib/python2.7 (from pwntools)
Downloading/unpacking mako (from pwntools)
  Running setup.py (path:/tmp/pip_build_root/mako/setup.py) egg_info for package mako

    warning: no files found matching '*.xml' under directory 'examples'
    warning: no files found matching '*.mako' under directory 'examples'
    warning: no files found matching 'ez_setup.py'
    no previously-included directories found matching 'doc/build/output'
Downloading/unpacking pyelftools (from pwntools)
  Running setup.py (path:/tmp/pip_build_root/pyelftools/setup.py) egg_info for package pyelftools

Downloading/unpacking capstone==2.1 (from pwntools)
  Downloading capstone-2.1.tar.gz
  Running setup.py (path:/tmp/pip_build_root/capstone/setup.py) egg_info for package capstone

Downloading/unpacking ropgadget (from pwntools)
  Downloading ROPGadget-5.4.tar.gz
  Running setup.py (path:/tmp/pip_build_root/ropgadget/setup.py) egg_info for package ropgadget

Requirement already satisfied (use --upgrade to upgrade): pyserial in /usr/lib/python2.7/dist-packages (from pwntools)
Requirement already satisfied (use --upgrade to upgrade): requests in /usr/lib/python2.7/dist-packages (from pwntools)
Downloading/unpacking psutil (from pwntools)
  Running setup.py (path:/tmp/pip_build_root/psutil/setup.py) egg_info for package psutil

    warning: no previously-included files matching '*' found under directory 'docs/_build'
Requirement already satisfied (use --upgrade to upgrade): pycrypto>=2.1,!=2.4 in /usr/lib/python2.7/dist-packages (from paramiko->pwntools)
Downloading/unpacking ecdsa>=0.11 (from paramiko->pwntools)
Downloading/unpacking MarkupSafe>=0.9.2 (from mako->pwntools)
  Downloading MarkupSafe-0.23.tar.gz
  Running setup.py (path:/tmp/pip_build_root/MarkupSafe/setup.py) egg_info for package MarkupSafe

Installing collected packages: pwntools, paramiko, mako, pyelftools, capstone, ropgadget, psutil, ecdsa, MarkupSafe
  Running setup.py install for pwntools

    Installing unhex script to /usr/local/bin
    Installing elfpatch script to /usr/local/bin
    Installing constgrep script to /usr/local/bin
    Installing cyclic script to /usr/local/bin
    Installing hex script to /usr/local/bin
    Installing phd script to /usr/local/bin
    Installing elfdiff script to /usr/local/bin
    Installing shellcraft script to /usr/local/bin
    Installing disasm script to /usr/local/bin
    Installing asm script to /usr/local/bin
  Running setup.py install for mako

    warning: no files found matching '*.xml' under directory 'examples'
    warning: no files found matching '*.mako' under directory 'examples'
    warning: no files found matching 'ez_setup.py'
    no previously-included directories found matching 'doc/build/output'
    Installing mako-render script to /usr/local/bin
  Running setup.py install for pyelftools
    changing mode of build/scripts-2.7/readelf.py from 644 to 755

    changing mode of /usr/local/bin/readelf.py to 755
  Running setup.py install for capstone

  Running setup.py install for ropgadget
    changing mode of build/scripts-2.7/ROPgadget from 644 to 755

    changing mode of /usr/local/bin/ROPgadget to 755
  Running setup.py install for psutil
    building '_psutil_linux' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPSUTIL_VERSION=221 -I/usr/include/python2.7 -c psutil/_psutil_linux.c -o build/temp.linux-x86_64-2.7/psutil/_psutil_linux.o
    psutil/_psutil_linux.c:12:20: fatal error: Python.h: No such file or directory
     #include <Python.h>
                        ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    Complete output from command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/psutil/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-CbWtQT-record/install-record.txt --single-version-externally-managed --compile:
    running install

running build

running build_py

creating build

creating build/lib.linux-x86_64-2.7

creating build/lib.linux-x86_64-2.7/psutil

copying psutil/_pswindows.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_pssunos.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_pslinux.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_psbsd.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_psposix.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_common.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_psosx.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/_compat.py -> build/lib.linux-x86_64-2.7/psutil

copying psutil/__init__.py -> build/lib.linux-x86_64-2.7/psutil

running build_ext

building '_psutil_linux' extension

creating build/temp.linux-x86_64-2.7

creating build/temp.linux-x86_64-2.7/psutil

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -DPSUTIL_VERSION=221 -I/usr/include/python2.7 -c psutil/_psutil_linux.c -o build/temp.linux-x86_64-2.7/psutil/_psutil_linux.o

psutil/_psutil_linux.c:12:20: fatal error: Python.h: No such file or directory

 #include <Python.h>

                    ^

compilation terminated.

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_root/psutil/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-CbWtQT-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip_build_root/psutil
Storing debug log for failure in /root/.pip/pip.log
STDERR: 
---- End output of pip install pwntools ----
Ran pip install pwntools returned 1
```

**Comment by InfamousSYN** [comment]
http://stackoverflow.com/questions/21530577/fatal-error-python-h-no-such-file-or-directory

**Comment by InfamousSYN** [comment]
Fixed in version 2.0.1
